OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -63,6 +63,8 @@ public class ArmamentInfo : ConditionalTraitInfo, Requires<AttackBaseInfo>
 		[Desc("Use multiple muzzle images if non-zero")]
 		public readonly int MuzzleSplitFacings = 0;
 
+		public readonly bool ShootAtClosestTargetablePosition = false;
```

Comment:
Right now all buildings have the same "health shape", which is the size of the actor that is considered for direct hits and damage falloff.  Ignoring armor, a shell that falls 1 cell away from the middle of a silo (i.e. half a cell outside its footprint) will do the same amount of damage as a cell that falls 1 cell away from the middle of a construction yard (half a cell *inside* its footprint).

You can see these shapes directly by turning on the "show combat geometry" in the debug menu.  The yellow circles are the health shapes.

Fixing the health shape to match the real footprint of the actor means that there is a larger area that inaccurate weapons can fall in and deal full damage, and the area that shells can hit and still do partial damage is larger.